compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtInterfaces.td

```diff
@@ -40,4 +41,98 @@ def LinalgExtInterface : OpInterface<"LinalgExtOp"> {
 }
 
+// Interface that allows for fusion of both LinalgExt and Linalg ops.
+def LinalgFusionInterface : OpInterface<"LinalgFusionOpInterface", [DestinationStyleOpInterface]> {
+  let methods = [
+    //===------------------------------------------------------------------===//
+    // Interface methods for fusion. 
+    // For linalg ops, these methods are already implemented
+    //===------------------------------------------------------------------===//
+      InterfaceMethod<
+      /*desc=*/[{
+        Return the number of parallel loops.
+      }],
+      /*retTy=*/"unsigned",
+      /*methodName=*/"getNumParallelLoops",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return llvm::count($_op.getLoopIteratorTypes(), utils::IteratorType::parallel);
+      }]
+    >,
+
+    InterfaceMethod<
+      /*desc=*/[{
+        Return the total number of loops.
+      }],
+      /*retTy=*/"unsigned",
+      /*methodName=*/"getNumLoops",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return $_op.getLoopIteratorTypes().size(); 
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Return the static loop ranges.
+      }],
+      /*retTy=*/"SmallVector<int64_t, 4>",
+      /*methodName=*/"getStaticLoopRanges",
+      /*args=*/(ins),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+          SmallVector<int64_t, 4> loopRanges;
+          llvm::for_each($_op.getOperands(), [&](Value operand) {
+          if (auto shapedType = dyn_cast<ShapedType>(operand.getType())) {
+            llvm::append_range(loopRanges, shapedType.getShape());
+            }
+          });
+          return loopRanges;
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Return the indexing map for an op's `result`.
+      }],
+      /*retTy=*/"AffineMap",
+      /*methodName=*/"getIndexingMapMatchingResult",
+      /*args=*/(ins "OpResult":$result),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        assert(result.getOwner() == $_op);
+        auto indexingMaps =
+          $_op.getIndexingMaps().template getAsValueRange<AffineMapAttr>();
+        return *(indexingMaps.begin() + $_op.getNumDpsInputs() +
+                 result.getResultNumber());
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Return the input or output indexing map for `opOperand`.
+      }],
+      /*retTy=*/"AffineMap",
+      /*methodName=*/"getMatchingIndexingMap",
+      /*args=*/(ins "OpOperand*":$opOperand),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        assert(opOperand->getOwner() == $_op);
+        auto indexingMaps =
+          $_op.getIndexingMaps().template getAsValueRange<AffineMapAttr>();
+        return *(indexingMaps.begin() + opOperand->getOperandNumber());
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        Return the indexing maps for this op.
+      }],
+      /*retTy=*/"ArrayAttr",
```

Comment:
Instead of an `ArrayAttr` make this method return a `SmallVector<AffineMap>`. I think we need two methods
1) `getIndexingMapsForOperands` which would return as many indexing maps as the number of operands
2) `getIndexingMapForResults` which would return as many indexing maps as the number of results.

If the indexing maps cannot be determined for an operand/result, the corresponding value can be nullptr.

You can use these in all the default implementation methods above.

Reply:
I'm not quite sure what you meant by returning nullptr, but a vector of optionals seemed like what you were saying